packages/webview_flutter/webview_flutter_wkwebview/lib/src/v4/src/webkit_webview_controller.dart

```diff
@@ -35,54 +35,62 @@ class WebKitWebViewControllerCreationParams
   }) : this(webKitProxy: webKitProxy);
 
   final WKWebViewConfiguration _configuration;
+
+  /// Handles constructing objects and calling static methods for the WebKit
+  /// native library.
+  @visibleForTesting
+  final WebKitProxy webKitProxy;
 }
 
 /// An implementation of [PlatformWebViewController] with the WebKit api.
 class WebKitWebViewController extends PlatformWebViewController {
   /// Constructs a [WebKitWebViewController].
-  WebKitWebViewController(
-    PlatformWebViewControllerCreationParams params, {
-    @visibleForTesting WebKitProxy webKitProxy = const WebKitProxy(),
-  }) : super.implementation(params is WebKitWebViewControllerCreationParams
+  WebKitWebViewController(PlatformWebViewControllerCreationParams params)
+      : super.implementation(params is WebKitWebViewControllerCreationParams
             ? params
             : WebKitWebViewControllerCreationParams
                 .fromPlatformWebViewControllerCreationParams(params)) {
-    final WeakReference<WebKitWebViewController> weakThis =
-        WeakReference<WebKitWebViewController>(this);
-    _webView = webKitProxy.createWebView(
-      (params as WebKitWebViewControllerCreationParams)._configuration,
+    webView.addObserver(
+      webView,
+      keyPath: 'estimatedProgress',
+      options: <NSKeyValueObservingOptions>{
+        NSKeyValueObservingOptions.newValue,
+      },
+    );
+  }
+
+  /// The WebKit WebView being controlled.
+  late final WKWebView webView = withWeakRefenceTo(this, (
```

Comment:
Why do we need to make this public?

Reply:
`WebViewWidget` needs this parameter and it's in a separate file. I was actually debating on how to handle this. We aren't at the point where we want to expose the wrapped API, so I might move the `WebKitWebViewController` and `WebKitWebViewWidget` in to the same file?

Reply:
Sounds reasonable; based on my recent discussions with Hixie it sounds like that's the preferred solution for this kind of thing.